ops/ansible/roles/bfd-server/templates/bfd-server.sh.j2

```diff
@@ -33,6 +33,77 @@ export NEW_RELIC_ENVIRONMENT='{{ data_server_new_relic_environment }}'
 {% endif %}
 {% endif %}
 
+{% if data_server_startup_testing is defined and data_server_startup_testing is sameas true %}
+# Begin smoke testing startup routine
+STARTUP_TESTING_RETRIES='3'
+STARTUP_TESTING_BOOT_TIMEOUT='30'
+STARTUP_TESTING_REQ_TIMEOUT='15'
+STARTUP_TESTING_REQ_BACKOFF_TIMEOUT='10'
+STARTUP_TESTING_BENE_ID='-201'
+
+##
+# Prints out the specified message.
+#
+# Params:
+# * $1: the message to log
+##
+log() {
+  echo "$(date +%F\ %T): $1"
+}
+
+##
+# Verifies that the specified endpoint/query can be queried via curl without error and produced the expected output.
+#
+# Params:
+# * $1: the full URL to query via curl
+# * $2: the value to confirm is present in the curl output; if not found, then the query should be regarded as unsuccessful
+##
+check_endpoint() {
+  set -o pipefail
+  curl --max-time $STARTUP_TESTING_REQ_TIMEOUT --silent --insecure --cert-type pem --cert "{{ data_server_dir }}/bluebutton-backend-test-data-server-client-test-keypair.pem" "$1" | grep "$2"
+}
+
+##
+# Runs test queries to verify that the server is working as expected. If not, issues a 'kill -INT' on the server's process to shut it down.
+##
+stop_service_if_failing() {
+  log "Waiting $STARTUP_TESTING_BOOT_TIMEOUT seconds for initial startup"
+  sleep $STARTUP_TESTING_BOOT_TIMEOUT
+
+  for I in $(seq $STARTUP_TESTING_RETRIES); do
+    log "Checking metadata endpoint"
+    STARTUP_TESTING_CHECK_METADATA=$(check_endpoint "https://localhost:7443/v1/fhir/metadata?_format=application%2Fjson%2Bfhir" "status")
+    STARTUP_TESTING_CHECK_METADATA_EXIT=$?
+
+    log "Checking database endpoint"
+    STARTUP_TESTING_CHECK_DATABASE=$(check_endpoint "https://localhost:7443/v1/fhir/Coverage?beneficiary=$STARTUP_TESTING_BENE_ID&_format=application%2Fjson%2Bfhir" "id")
```

Comment:
This is testing the Coverage resource end-point, which includes the database connection, but has more. I suggest updating the log comment.